.github/workflows/docker-and-ks8.yml

```diff
@@ -0,0 +1,128 @@
+          python-version: 3.8.10
+
+      - name: Perform the deployment of the test environment.
+        uses: ./.github/actions/perform-deployment
+        with:
+          deployment-name: "test_environment"
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+
+      - name: Start minikube for the test of the kubernetes build.
+        if: matrix.image-type == 'k8s'
+        shell: bash
+        run: |
+          minikube start
+
+      - name: Run image test.
+        env:
+            SCALYR_API_KEY: ${{ secrets.SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN }}
+        run: |
+          python3 tests/package_tests/run_package_test.py run-package-test ${{ matrix.image-type}}_test
+
+# TODO need to add READ_API_KEY secret from circleci
+#      - name: Run docker smoke tests
+#        if: startsWith( matrix.image-type, 'docker-')
+#        env:
+#          SCALYR_API_KEY: ${{ secrets.SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN }}
+#          READ_API_KEY: # Need to add to github
+#
+#        run: |
+#          source ./.circleci/smoketest_docker.sh scalyr/scalyr-agent-ci-unittest:4 json 150
+
+
+  publish:
+    needs: [test]
+    name: Publish image.
+    if: ${{ github.ref == 'refs/heads/master' || github.ref_type == 'tag' }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        image-type: [docker-json, docker-syslog k8s]
+
+
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+
+      - name: Push image from master branch.
+        if: ${{ github.ref == 'refs/heads/master' }}
+        run: |
+
+          python3 build_package_new.py ${{ matrix.image-type }} \
+            --push \
+            --registry \
+            localhost:5000 \
+            --tag ${{ github.sha }}
+
+      - name: Push image using master branch commit hash.
+        if: ${{ github.ref_type == 'tag' }}
+        run: |
+
+          if [[ "${{ github.ref_name }}" == "v$(cat VERSION)" ]]; then
+            # if the this is a "release" tag with format "v*.*.*", then use version as tag
+            # also add the 'latest' tag.
+            tags="--tag $(cat VERSION) --tag latest"
+          else
+            # Just use GitHub tag as image tag.
+            tags="--tag ${{ github.ref_name }}"
+          fi
+
+          python3 build_package_new.py ${{ matrix.image-type }} \
```

Comment:
At what point do we plan to switch to Dockerhub?

I think once the tests are running + passing it should likely be fine to switch to Docker hub for the master dev builds.